### DISCUSSION TITLE
Combobox eating default Home/End key behavior

### DISCUSSION THREAD

**Discussion by shyamajp** [twilio-labs/paste#2100] [Q&A] [answered]
Hello Paste team!

## Problem
We are using `Combobox` as select item with search, but Home/End key doesn't work when selecting cursor-to-end text with Shift + End. Is there a bypass for this behavior?

### Customer report (modified)
```
the Home and End keys don't work in the text part of the Combobox fields.

I've typed some prefix and selected an item. Then I realise I've selected the wrong one (say I've selected "Test API" when I intended "Test Another API"). So I put the cursor after "Test" and press Shift+End to select to the end of the field, so I can press Del (or BkSp). The cursor doesn't move. Left/Right work fine.

I suspect that it's because the drop-down part of the field is eating Home/End (meaning Top/Bottom). It shouldn't do that when the drop-down is closed and focus is in the text field.
```

Code sandbox attached below, but it is basically same as [Official Combobox Example](https://paste.twilio.design/components/combobox/#autocomplete-combobox)

https://codesandbox.io/s/compassionate-waterfall-e3lk7?file=/src/index.tsx

Thank you very much in advance.

**Comment by TheSisb** [comment, marked as answer]
Hello @shyamajp,

Our code doesn't block this behavior, so I checked Downshift, which is the library we built our Combobox around. It looks like this behavior [is intended](https://github.com/downshift-js/downshift/issues/1097). That said, they have [some docs on how to amend the behavior](https://github.com/downshift-js/downshift#customizing-handlers) to fit your needs.

I was able to make a fix with our component here:  https://codesandbox.io/s/vibrant-fire-pmjkp?file=/src/index.tsx
Hope this helps! Let us know if you have any other questions.
